pydantic_ai_slim/pydantic_ai/_tool_execution.py

```
@@ -240,12 +251,19 @@ def __post_init__(self) -> None:
             # including `'unknown'` (hallucinated) ones, using the `'skip'` sentinel for any call that
             # was already handled in a prior step. The check below relies on that convention.
             self.executable_function_kinds = ('function', 'unknown', 'external', 'unapproved')
+            eligible_calls = [
+                call for call, kind in zip(self.tool_calls, call_kinds) if kind in self.executable_function_kinds
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)